ui/admin/app/routes/scopes/scope/credential-stores/credential-store/credential-libraries/new.js

```diff
@@ -0,0 +1,45 @@
+import Route from '@ember/routing/route';
+
+export default class ScopesScopeCredentialStoresCredentialStoreCredentialLibrariesNewRoute extends Route {
+  // =methods
+
+  /**
+   * Creates a new unsaved credential library in current credential store.
+   * @return {CredentialLibraryModel}
+   */
+  model() {
+    const { id: credential_store_id } = this.modelFor(
+      'scopes.scope.credential-stores.credential-store'
+    );
+    return this.store.createRecord('credential-library', {
+      credential_store_id,
+    });
+  }
+
+  /**
+   * Renders new credential library specific templates for header, navigation,
+   * and action page sections.
+   * @override
+   */
+  renderTemplate() {
+    super.renderTemplate(...arguments);
+
+    this.render(
+      'scopes/scope/credential-stores/credential-store/credential-libraries/new/-header',
+      {
+        into: 'scopes/scope/credential-stores/credential-store',
+        outlet: 'header',
+      }
+    );
+
+    this.render('_empty', {
```

Comment:
This is a question more than a comment. I didn't see these before. Is this a way to could define the `renderTemplate` before having the template file ready?

Reply:
This is going to render an [_empty.hbs](https://github.com/hashicorp/boundary-ui/blob/main/ui/admin/app/templates/_empty.hbs) template file that's available at `templates` root. It was created in a previous iteration to provide us with a way to clear named outlets as needed. Render method does complain loudly when attempting to render a file that doesn't exist.

Reply:
Awesome! thanks :)